nexus/types/src/deployment.rs

```diff
@@ -1376,6 +1371,21 @@ impl FromStr for ExpectedVersion {
     }
 }
 
+/// Describes the expected active RoT slot, and the version we expect to find for it
+#[derive(
+    Clone, Debug, Eq, PartialEq, JsonSchema, Deserialize, Serialize, Diffable,
+)]
+pub struct ExpectedActiveRotSlot {
+    pub slot: RotSlot,
+    pub version: ArtifactVersion,
+}
+
+impl ExpectedActiveRotSlot {
+    pub fn slot(&self) -> &RotSlot {
```

Comment:
Tiny nit - it seems a little weird to have a method for slot but not one for version. I don't have a strong feeling either way; I'd probably remove this method and just use the `slot` field directly like we do for version. But either is fine - should just be consistent.